src/Stateless/MethodDescription.cs

```diff
@@ -0,0 +1,51 @@
+﻿using System;
+
+namespace Stateless
+{
+    internal class MethodDescription
+    {
+        readonly string _description;                     // _description can be null if user didn't specify a description
+
+        public enum Timing
+        {
+            Synchronous,
+            Asynchronous
+        }
+        Timing _timing;
+
+        public static MethodDescription Create(Delegate method, string description, Timing timing = Timing.Synchronous)
+        {
+            return new MethodDescription(method?.TryGetMethodName(), description, timing);
+        }
+
+        private MethodDescription(string methodName, string description, Timing timing)      // description can be null if user didn't specify a description
+        {
+            MethodName = methodName;
+            _description = description;
+            _timing = timing;
+        }
+
+        /// <summary>
+        /// The name of the invoked method.  If the method is a lambda or delegate, the name will be a compiler-generated
+        /// name that is often not human-friendly (e.g. "(.ctor)b__2_0" except with angle brackets instead of parentheses)
+        /// </summary>
+        internal string MethodName { get; private set; }
```

Comment:
Yes they can ... but I'm not sure it increases the readability of the code.  It's an interesting problem, you might consider it a limitation of the language.  If you read a method that says it's public, you don't know what that means until you look up at the class definition too.  I guess I went with defining the property as internal so it was more obvious what its true accessibility is.